pkgs/applications/audio/stargate/default.nix

```diff
@@ -0,0 +1,167 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, substituteAll
+, alsa-lib
+, autoconf
+, automake
+, ffmpegSupport ? true
+, ffmpeg
+, fftw
+, fftwSinglePrec
+, file
+, gdb
+, gettext
+, jq
+, lameSupport ? true
+, lame
+, libsbsms
+, libsndfile
+, pkg-config
+, portaudio
+, portmidi
+, python3
+, python3Packages
+, rubberband
+, stargate-libcds
+, vorbisToolsSupport ? true
+, vorbis-tools
+, wrapQtAppsHook
+}:
+
+let
+  python3CommonInputs = python3.withPackages
+    (ps: with ps; [
+      jinja2
+      mido
+      mutagen
+      numpy
+      psutil
+      pymarshal
+      pyqt5
+      pyyaml
+      wavefile
+      wheel
+    ]);
+in
+stdenv.mkDerivation rec {
```

Comment:
I think we should use the `finalAttrs` pattern here.

Let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe